### PR TITLE
Milestone 5: Vet Clinic | Database performance audit

### PR DESCRIPTION
**Pull request by michael-duke**
## Vet Clinic milestone 5: Database Performance Audit

In this milestone @kendoriddy and @michael-duke  have :
- Understand what can impact database performance.
- Prepared the data to have an `Execution time: X ms` > `1000ms `ours was `1251ms`
- Optimizes some slow queries in `vet_clinic` database.
- Used `CREATE INDEX` to enhance query performance execution time
- Used `EXPLAIN ANALYZE` to understand query data

⚠️ **_Kindly see the updated amended changes below under the previous reviewer_**
## Explain Analyze Results `BEFORE`:
before-part 1
screen part1-b

## Explain Analyze Results `AFTER`:
After

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -91,4 +97,51 @@ CREATE TABLE visits (
...
+    date_of_visit DATE NOT NULL
+);
+
+-- Add an email column to your owners table
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
+
+-- Create index to enhance performance
+CREATE INDEX visits_animal_id ON visits(animal_id);
+CREATE INDEX visits_vet_id ON visits(vet_id);
+CREATE INDEX owner_email ON owners(email);
```
Comment (ShadyShawkat) [inline schema.sql]: - [ ] Kindly make sure that the execution time of each query before creating the indexes is at least 1000ms, as per [this](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_performance_audit.md#:~:text=If%20you%20get%20Execution%20time%3A%20X%20ms%20and%20X%20%3C%201000%3A%20please%20go%20back%20to%20point%203.%20and%20repeat%20until%20you%20get%20a%20value%20bigger%20than%201000ms) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (michael-duke) [reply schema.sql]: @ShadyShawkat we have attached the screenshots for the 'BEFORE' creating the index Are you referring to the 2 pictures we have attached

Comment (ShadyShawkat) [reply schema.sql]: @michael-duke, yes the 'BEFORE` images should have execution times for each query that surpasses 1000ms.

Comment (michael-duke) [reply schema.sql]: Thanks @ShadyShawkat  we will adjust and make a request, happy coding 😄 🚀

**Comment by michael-duke** [comment]
### Dear Reviewer, kindly find the `requested changes below 😄:

## Explain Analyze Results `BEFORE` amended after review:
image

## Explain Analyze Results `AFTER`  amended after review:
image

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @michael-duke & @kendoriddy, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by michael-duke** [comment]
Thanks @sja-the-dude for your feedback and happy coding 😄 🚀

**Pull request merged**